### ISSUE TITLE
Does not check the lack of NSEC3 records ?

### ISSUE THREAD

**Issue by bortzmeyer** [zonemaster/zonemaster#475]
Among the three name servers of .sx, one does not return NSEC3 records:

```
% dig @b.ns.sx A hfgygtfgsqdghfkjsdhgsdkffghd.sx

; <<>> DiG 9.10.3-P4-Debian <<>> @b.ns.sx A hfgygtfgsqdghfkjsdhgsdkffghd.sx
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 43115
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 2, ADDITIONAL: 1
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 4096
;; QUESTION SECTION:
;hfgygtfgsqdghfkjsdhgsdkffghd.sx. IN A

;; AUTHORITY SECTION:
sx.         86400 IN SOA a.ns.sx. sx.openregistry.com. (
                15437      ; serial
                14400      ; refresh (4 hours)
                3600       ; retry (1 hour)
                604800     ; expire (1 week)
                86400      ; minimum (1 day)
                )
sx.         86400 IN RRSIG SOA 7 1 86400 (
                20170415212004 20161017212004 39185 sx.
                YidMh+bq2M6rir6SYosVfGmGaX6OPnYSotsYQaL/q/1M
                RSKXnoLtwYroqxGaLvZNSs76Re7yj3jp9YxtZrZogSl6
                rcLVe+MQb72kdgIwJhFZ+3TA+qHNhkVYPzNdqBxF52h1
                04mdbpfiqzQ5PQkJJ2GO4fkeyN0ZRqhnpqejjaE= )

;; Query time: 42 msec
;; SERVER: 80.92.90.160#53(80.92.90.160)
;; WHEN: Tue Oct 18 09:29:06 CEST 2016
;; MSG SIZE  rcvd: 282
```

The others do:

```
% dig @a.ns.sx A hfgygtfgsqdghfkjsdhgsdkffghd.sx

; <<>> DiG 9.10.3-P4-Debian <<>> @a.ns.sx A hfgygtfgsqdghfkjsdhgsdkffghd.sx
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 62272
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 8, ADDITIONAL: 1
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 4096
;; QUESTION SECTION:
;hfgygtfgsqdghfkjsdhgsdkffghd.sx. IN A

;; AUTHORITY SECTION:
sx.         86400 IN SOA a.ns.sx. sx.openregistry.com. (
                15437      ; serial
                14400      ; refresh (4 hours)
                3600       ; retry (1 hour)
                604800     ; expire (1 week)
                86400      ; minimum (1 day)
                )
sx.         86400 IN RRSIG SOA 7 1 86400 (
                20170415212004 20161017212004 39185 sx.
                YidMh+bq2M6rir6SYosVfGmGaX6OPnYSotsYQaL/q/1M
                RSKXnoLtwYroqxGaLvZNSs76Re7yj3jp9YxtZrZogSl6
                rcLVe+MQb72kdgIwJhFZ+3TA+qHNhkVYPzNdqBxF52h1
                04mdbpfiqzQ5PQkJJ2GO4fkeyN0ZRqhnpqejjaE= )
N32PJU8P60HH9B9KC79Q869VNAC9NFHS.sx. 86400 IN NSEC3 1 0 10 4321 (
                N3Q833IAF3FHN1BP8QOD6CC68EPQCUH4
                NS SOA RRSIG DNSKEY NSEC3PARAM )
N32PJU8P60HH9B9KC79Q869VNAC9NFHS.sx. 86400 IN RRSIG NSEC3 7 2 86400 (
                20170415212004 20161017212004 39185 sx.
                stDo9xqihJeSJw8sXeqeWx3pkCqZo8MWWozkp/zbGENX
                EPuxpyS4fUvV4sLiVRBp4GSjpX9Cq+q1GjCh9KeLa8Uc
                +PK5LWcb962ae7PKFUCZ3ALEIiBVh3irGIh/RT3TFGBf
                BoD40pfw2hdyvNLSTIFGoAuxxETS4b/y2I8HOoY= )
IPBUE7LAJJAT96320E3S0B5I9SGGEE9M.sx. 86400 IN NSEC3 1 0 10 4321 (
                IPL92RQS9HR3LJ6GDJBPNP529ALUDSBU
                NS )
IPBUE7LAJJAT96320E3S0B5I9SGGEE9M.sx. 86400 IN RRSIG NSEC3 7 2 86400 (
                20170415212004 20161017212004 39185 sx.
                XCcMHKvcOeDkEkau+GQzoU47yWmDIHzIxr3IvNdznyZX
                /boJNtsNlYMJcm2GY0vJqj9wJy6om9FtBQ7533QSoCBE
                8L3jMLzlJjRMQuFFM1vqAsfqlR75//U1Bd86zB0/FRwf
                6upn8QLVGcWQ8mEdYXkp2nfAbO6Jp1/ZY9g2ZPg= )
RJ6N95LAQKR3B6CLHSPP1Q9PSA5CH7T5.sx. 86400 IN NSEC3 1 0 10 4321 (
                RJHSF98P8P2SERO9ULBGF7GPKH0LJSHL
                NS )
RJ6N95LAQKR3B6CLHSPP1Q9PSA5CH7T5.sx. 86400 IN RRSIG NSEC3 7 2 86400 (
                20170415212004 20161017212004 39185 sx.
                ty8ogSPvrQWfY1iaEPpDfpkpNEfXm7JkzTopgv/3fOvI
                5rlcqnMqgAPqHp8YYNE1Sq5EdH1FkHJKv3uyLePEVZ+9
                SExLu1joQmG0PCvMVCJ/lyRBDSldDVBmK4XWZawzYdKl
                MQNSsgIXP+CSTLtJBrH6P9h4awRpXZLlef/peKs= )

;; Query time: 28 msec
;; SERVER: 2a02:2790::600d:deed#53(2a02:2790::600d:deed)
;; WHEN: Tue Oct 18 09:29:43 CEST 2016
;; MSG SIZE  rcvd: 1002
```

Unfortunately, Zonemaster does not catch the error https://zonemaster.net/test/27fc77fa8bb4c806

DNSviz does http://dnsviz.net/d/cf746c21f68a023a4a8daf78.sx/dnssec/

**Comment by matsduf** [comment]
Moved to zonemaster-engine repo https://github.com/dotse/zonemaster-engine/issues/237

**Comment by matsduf** [comment]
Create a new Test Case covering this problem.

**Comment by matsduf** [comment]
Can specification be updated to capture this?

**Comment by sandoche2k** [comment]
Should update https://github.com/zonemaster/zonemaster/blob/master/docs/specifications/tests/DNSSEC-TP/dnssec10.md to cover #475, #496, #497

**Comment by matsduf** [comment]
The first server, in the example, shows no sign of DNSSEC, so the issue presented here is more about consistent DNSSEC in all servers.

We could update DNSSEC10 in two aspects:
* All servers are checked and it is noted for each if NSEC, NSEC3 or none is returned.
* The results from all servers are compared and an error is outputted if inconsistent.

**Comment by matsduf** [comment]
Resolved by #786.
